### PR TITLE
Froggo local teams

### PR DESCRIPTION
**Pull request by pfarellano**
En este PR están los commits de la creación de los modelos, controladores, serializadores y cosas básicas para definir los equipos locales de froggo :) Solo back-end.

### CODE REVIEW

File: app/models/froggo_team.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class FroggoTeam < ApplicationRecord
+  has_many :froggo_team_memberships
```
Comment (monkeyci) [inline app/models/froggo_team.rb:2]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Rails/HasManyOrHasOneDependent: Specify a :dependent option.

File: app/models/github_user.rb [hunk 2]
```diff
@@ -10,6 +10,9 @@ class GithubUser < ApplicationRecord
   has_many :organization_memberships
   has_many :organizations, through: :organization_memberships, source: :organization
 
+  has_many :froggo_team_memberships
```
Comment (monkeyci) [inline app/models/github_user.rb:13]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Rails/HasManyOrHasOneDependent: Specify a :dependent option.

File: app/models/organization.rb [hunk 3]
```diff
@@ -5,6 +5,7 @@ class Organization < ApplicationRecord
   has_many :hooks, as: :resource
   has_many :organization_memberships
   has_many :members, through: :organization_memberships, source: :github_user
+  has_many :froggo_teams
```
Comment (monkeyci) [inline app/models/organization.rb:8]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Rails/HasManyOrHasOneDependent: Specify a :dependent option.

File: spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+RSpec.describe Api::V1::FroggoTeamMembershipsController, type: :controller do
+  let(:user) { create(:github_user, login: "user_login") }
+  let(:platanus_org) { create(:organization, login: "platanus", members: [user]) }
+  let!(:platanus_team) { create(:froggo_team, name: "test_name", organization: platanus_org) }
+  let(:buda_org) { create(:organization) }
+  let!(:buda_team) { create(:froggo_team, name: "test_name", organization: buda_org) }
+  let(:user_1) { create(:github_user, login: "user1_login") }
+  let!(:join) { create(:froggo_team_membership, github_user: user_1, froggo_team: platanus_team) }
+  let(:github_session) do
+    double(
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb:12]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/VerifiedDoubles: Prefer using verifying doubles over normal doubles.

File: spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      token: "token",
+      organizations: [{ id: platanus_org.gh_id, role: "member" }],
+      user: user
+    )
+  end
+
+  before do
+    expect(subject).to receive(:authenticate_github_user).and_return(true)
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb:20]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/ExpectInHook: Do not use expect in before hook

Comment (monkeyci) [inline spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb:20]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/SubjectStub: Do not stub methods of the object under test.

Comment (monkeyci) [inline spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb:20]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/NamedSubject: Name your test subject if you need to reference it explicitly.

Comment (monkeyci) [inline spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb:20]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/MessageSpies: Prefer have_received for setting message expectations. Setup subject as a spy using allow or instance_spy.

File: spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    allow(subject).to receive(:github_session).and_return(github_session)
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb:21]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/SubjectStub: Do not stub methods of the object under test.

Comment (monkeyci) [inline spec/controllers/api/v1/froggo_team_memberships_controller_spec.rb:21]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/NamedSubject: Name your test subject if you need to reference it explicitly.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,109 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::FroggoTeamsController, type: :controller do
+  let(:user) { create(:github_user) }
+  let(:platanus_org) { create(:organization, login: "platanus", members: [user]) }
+  let!(:platanus_team) { create(:froggo_team, name: "test_name", organization: platanus_org) }
+  let(:buda_org) { create(:organization) }
+  let!(:buda_team) { create(:froggo_team, name: "test_name", organization: buda_org) }
+  let(:github_session) do
+    double(
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:10]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/VerifiedDoubles: Prefer using verifying doubles over normal doubles.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+      token: "token",
+      organizations: [{ id: platanus_org.gh_id, role: "member" }],
+      user: user
+    )
+  end
+
+  before do
+    expect(subject).to receive(:authenticate_github_user).and_return(true)
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:18]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/ExpectInHook: Do not use expect in before hook

Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:18]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/SubjectStub: Do not stub methods of the object under test.

Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:18]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/NamedSubject: Name your test subject if you need to reference it explicitly.

Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:18]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/MessageSpies: Prefer have_received for setting message expectations. Setup subject as a spy using allow or instance_spy.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    allow(subject).to receive(:github_session).and_return(github_session)
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:19]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/SubjectStub: Do not stub methods of the object under test.

Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:19]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/NamedSubject: Name your test subject if you need to reference it explicitly.

File: spec/services/froggo_team_service_spec.rb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+describe FroggoTeamService do
+  def build(*_args)
+    described_class.new(*_args)
+  end
+
+  pending "describe what your service does here"
+end
```
Comment (monkeyci) [inline spec/services/froggo_team_service_spec.rb:9]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Layout/TrailingEmptyLines: Final newline missing.

File: app/services/froggo_team_service.rb [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    organization.froggo_teams.each do |team|
+      return false unless team.name != name
+    end
+    true
+  end
+
+  def validate_member(github_user, organization)
+    return false unless organization.members.include?(github_user)
+
+    true
```
Comment (jgmontoya) [inline app/services/froggo_team_service.rb:12]: Acá podrías poner simplemente `organization.members.include?(github_user)`

File: app/services/froggo_team_service.rb [hunk 12]
```diff
@@ -0,0 +1,14 @@
+class FroggoTeamService < PowerTypes::Service.new
```
Comment (jgmontoya) [inline app/services/froggo_team_service.rb:1]: Creo que estas validaciones deberían estar en el modelo en lugar de en un servicio

File: app/controllers/api/v1/froggo_teams_controller.rb [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+    end
+  end
+
+  def show
+    respond_with froggo_team
+  end
+
+  def update
+    froggo_team_service = FroggoTeamService.new
+    organization = Organization.find(froggo_team.organization.id)
```
Comment (jgmontoya) [inline app/controllers/api/v1/froggo_teams_controller.rb:28]: No necesitas hacer `Organization.find` porque el `froggo_team` ya "sabe" cual es su organization. Esta linea es equivalente a `organization = froggo_team.organization` con lo que te ahorras una query.

File: app/controllers/api/v1/froggo_team_memberships_controller.rb [hunk 14]
```diff
@@ -0,0 +1,40 @@
+class Api::V1::FroggoTeamMembershipsController < Api::V1::BaseController
+  before_action :authenticate_github_user
+  def create
+    froggo_team_service = FroggoTeamService.new
+    organization = froggo_team.organization
```
Comment (jgmontoya) [inline app/controllers/api/v1/froggo_team_memberships_controller.rb:5]: Acá lo hiciste al tiro! :smile:

File: config/routes.rb [hunk 15]
```diff
@@ -41,6 +41,8 @@
         'github_users#team_score'
       get 'teams/:team_id/users/:github_login/recommendations' =>
         'github_users#team_review_recommendations'
+      resources :froggo_teams, only: [:index, :show, :create, :destroy, :update]
```
Comment (jgmontoya) [inline config/routes.rb:44]: Porque los teams de froggo pertenecen siempre a una organización, creo que tiene más sentido anidar los recursos, así además puedes contar con tener el id de la organización en la ruta.

```ruby
resources :organizations do
  resources :froggo_teams, only: [:index, :show, :create, :destroy, :update]
end
```

Esto te genera rutas de este estilo: `/api/organizations/:organization_id/froggo_teams/:id`

File: app/controllers/api/v1/froggo_teams_controller.rb [hunk 16]
```diff
@@ -0,0 +1,63 @@
+class Api::V1::FroggoTeamsController < Api::V1::BaseController
+  before_action :authenticate_github_user
+  def index
+    render json: { froggo_teams: github_session.user.froggo_teams }
+  end
+
+  def create
+    froggo_team_service = FroggoTeamService.new
+    valid_member = froggo_team_service.validate_member(github_session.user, organization)
```
[the remark this answers is not in the document]
Comment (pfarellano) [reply app/controllers/api/v1/froggo_teams_controller.rb:9]: lo que me pasa con definir esto, es que los valid_member de distintos métodos reciben distintos argumentos, entonces por ejemplo en el método create, uso una variable local de la función como parámetro Organization, entonces no se que conviene ahí

File: app/controllers/api/v1/froggo_teams_controller.rb [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+    render(json: { error: "permission denied" }, status: :bad_request) && return unless valid_member
+
+    name_validated = froggo_team_service.validate_team_name(permitted_params[:name], organization)
+    if name_validated
```
[the remark this answers is not in the document]
Comment (pfarellano) [reply app/controllers/api/v1/froggo_teams_controller.rb:13]: me pasa lo mismo de arriba, en uno de los dos métodos que usan esta función, se define el parámetro  Organization en forma local :(

File: app/services/froggo_team_service.rb (hunk 12 above)
Comment (pfarellano) [reply app/services/froggo_team_service.rb:1]: a que validaciones te refieres? la del nombre y el miembro que pertenece a una organización?

File: app/controllers/api/v1/froggo_teams_controller.rb (hunk 16 above)
[the remark this answers is not in the document]
Comment (jgmontoya) [reply app/controllers/api/v1/froggo_teams_controller.rb:9]: Pero la organización que te interesa para estos efectos siempre es la misma: la organización a la cual el team pertenece. Cuando hagas el cambio de las rutas puedes cambiar un poco el método `organization`

```ruby
def organization
  @organization ||= Organization.find(permitted_params[:organization_id]) || froggo_team.organization
end
```

File: app/services/froggo_team_service.rb (hunk 12 above)
Comment (jgmontoya) [reply app/services/froggo_team_service.rb:1]: Sip

File: app/models/organization.rb (hunk 3 above)
Comment (monkeyci) [inline app/models/organization.rb:8]: Specify a :dependent option.

File: app/models/github_user.rb (hunk 2 above)
Comment (monkeyci) [inline app/models/github_user.rb:13]: Specify a :dependent option.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+      it { expect(response).to have_http_status(400) }
+    end
+
+    context "when the team was succesfuly destroyed" do
+      before do
+        delete :destroy, params: { id: platanus_team.id }, format: :json
+      end
+
+      it { expect(response).to have_http_status(204) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:106]: Prefer :no_content over 204 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+    end
+  end
+
+  describe '#destroy' do
+    context "when destroyer user is not from organization" do
+      before do
+        delete :destroy, params: { id: buda_team.id }, format: :json
+      end
+
+      it { expect(response).to have_http_status(400) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:98]: Prefer :bad_request over 400 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+    end
+
+    context "when everything is ok" do
+      before do
+        patch :update, params: { id: platanus_team.id, name: "new_name" }, format: :json
+      end
+
+      it { expect(response).to have_http_status(200) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:88]: Prefer :ok over 200 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+
+      it { expect(response).to have_http_status(400) }
+    end
+
+    context "when the name is already taken" do
+      before do
+        patch :update, params: { id: platanus_team.id, name: "test_name" }, format: :json
+      end
+
+      it { expect(response).to have_http_status(400) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:80]: Prefer :bad_request over 400 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
+    end
+  end
+
+  describe '#update' do
+    context "when updater user is not from organization" do
+      before do
+        patch :update, params: { id: buda_team.id, name: "test_name" }, format: :json
+      end
+
+      it { expect(response).to have_http_status(400) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:72]: Prefer :bad_request over 400 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 23]
```diff
@@ -0,0 +1,109 @@
...
+      end
+
+      it { expect(response).to have_http_status(200) }
+    end
+  end
+
+  describe '#show' do
+    it 'responds with ok' do
+      get :show, params: { id: platanus_team.id }, format: :json
+      expect(response).to have_http_status(200)
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:62]: Prefer :ok over 200 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 24]
```diff
@@ -0,0 +1,109 @@
...
+
+      it { expect(response).to have_http_status(400) }
+    end
+
+    context "when everything is ok" do
+      before do
+        post :create, params: { name: "new_name", org_id: platanus_org.id }, format: :json
+      end
+
+      it { expect(response).to have_http_status(200) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:55]: Prefer :ok over 200 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
+
+      it { expect(response).to have_http_status(400) }
+    end
+
+    context "when name already exists for organization" do
+      before do
+        post :create, params: { name: "test_name", org_id: platanus_org.id }, format: :json
+      end
+
+      it { expect(response).to have_http_status(400) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:47]: Prefer :bad_request over 400 to describe HTTP status code.

File: app/models/froggo_team.rb (hunk 1 above)
Comment (monkeyci) [inline app/models/froggo_team.rb:2]: Specify a :dependent option.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 26]
```diff
@@ -0,0 +1,109 @@
...
+    let(:organization) { platanus_org }
+
+    context "when creator user is not from organization" do
+      let(:organization) { buda_org }
+
+      before do
+        post :create, params: { name: "test_name", org_id: buda_org.id }, format: :json
+      end
+
+      it { expect(response).to have_http_status(400) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:39]: Prefer :bad_request over 400 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 27]
```diff
@@ -0,0 +1,109 @@
...
+
+  before do
+    expect(subject).to receive(:authenticate_github_user).and_return(true)
+    allow(subject).to receive(:github_session).and_return(github_session)
+  end
+
+  describe '#index' do
+    it 'responds with ok' do
+      get :index
+      expect(response).to have_http_status(200)
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:25]: Prefer :ok over 200 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 28]
```diff
@@ -0,0 +1,156 @@
...
+    end
+  end
+
+  describe '#remove_member' do
+    context "when the user who wants to delete a member from a team is not from the organization" do
+      before do
+        post :remove_member, params: { id: buda_team.id, github_login: user.login }, format: :json
+      end
+
+      it { expect(response).to have_http_status(400) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:145]: Prefer :bad_request over 400 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 29]
```diff
@@ -0,0 +1,156 @@
...
+    end
+
+    context "when the member who is going to be added is from the organization" do
+      before do
+        post :add_member, params: { id: p_team.id, github_login: user.login }, format: :json
+      end
+
+      it { expect(response).to have_http_status(200) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:135]: Prefer :ok over 200 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 30]
```diff
@@ -0,0 +1,156 @@
...
+
+      it { expect(response).to have_http_status(400) }
+    end
+
+    context "when the member who is going to be added is not from the organization" do
+      before do
+        post :add_member, params: { id: p_team.id, github_login: "new_user" }, format: :json
+      end
+
+      it { expect(response).to have_http_status(400) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:127]: Prefer :bad_request over 400 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 31]
```diff
@@ -0,0 +1,156 @@
...
+    end
+  end
+
+  describe '#add_member' do
+    context "when the user who wants to add a member is not from the team organization" do
+      before do
+        post :add_member, params: { id: buda_team.id, github_login: user.login }, format: :json
+      end
+
+      it { expect(response).to have_http_status(400) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:119]: Prefer :bad_request over 400 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 32]
```diff
@@ -0,0 +1,156 @@
...
+    end
+
+    context "when the team was succesfuly destroyed" do
+      before do
+        delete :destroy, params: { id: new_team.id }, format: :json
+      end
+
+      it { expect(response).to have_http_status(204) }
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:109]: Prefer :no_content over 204 to describe HTTP status code.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 33]
```diff
@@ -0,0 +1,156 @@
...
+    double(
+      token: "token",
+      organizations: [{ id: platanus_org.gh_id, role: "member" }],
+      user: user
+    )
+  end
+
+  before do
+    expect(subject).to receive(:authenticate_github_user).and_return(true)
+    allow(subject).to receive(:github_session).and_return(github_session)
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:22]: Name your test subject if you need to reference it explicitly.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 34]
```diff
@@ -0,0 +1,156 @@
...
+  let(:github_session) do
+    double(
+      token: "token",
+      organizations: [{ id: platanus_org.gh_id, role: "member" }],
+      user: user
+    )
+  end
+
+  before do
+    expect(subject).to receive(:authenticate_github_user).and_return(true)
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:21]: Do not use expect in before hook<br>Name your test subject if you need to reference it explicitly.<br>Prefer have_received for setting message expectations. Setup subject as a spy using allow or instance_spy.

File: spec/controllers/api/v1/froggo_teams_controller_spec.rb [hunk 35]
```diff
@@ -0,0 +1,156 @@
...
+  let(:user) { create(:github_user, login: "user_login") }
+  let(:platanus_org) { create(:organization, login: "platanus", members: [user]) }
+  let!(:p_team) { create(:froggo_team, name: "test_name", organization: platanus_org) }
+  let(:new_team) { create(:froggo_team, name: "test_name2", organization: platanus_org) }
+  let(:buda_org) { create(:organization) }
+  let!(:buda_team) { create(:froggo_team, name: "test_name", organization: buda_org) }
+  let(:user_1) { create(:github_user, login: "user1_login") }
+  let!(:join) { create(:froggo_team_membership, github_user: user_1, froggo_team: p_team) }
+  let(:github_session) do
+    double(
```
Comment (monkeyci) [inline spec/controllers/api/v1/froggo_teams_controller_spec.rb:13]: Prefer using verifying doubles over normal doubles.

**Pull request merged**